esmvaltool/preprocessor/_mask.py

```diff
@@ -17,20 +19,109 @@
 logger = logging.getLogger(__name__)
 
 
-def mask_landocean(cube, land_mask=None, ocean_mask=None):
-    """Mask land or ocean"""
-    if land_mask or ocean_mask:
-        raise NotImplementedError
-
+def _check_dims(cube, mask_cube):
+    """Check for same dims for mask and data"""
+    x_dim = cube.coord('longitude').points.ndim
+    y_dim = cube.coord('latitude').points.ndim
+    mx_dim = mask_cube.coord('longitude').points.ndim
+    my_dim = mask_cube.coord('latitude').points.ndim
+    len_x = len(cube.coord('longitude').points)
+    len_y = len(cube.coord('latitude').points)
+    len_mx = len(mask_cube.coord('longitude').points)
+    len_my = len(mask_cube.coord('latitude').points)
+    if x_dim == mx_dim and y_dim == my_dim:
+        if len_x == len_mx and len_y == len_my:
+            logger.debug('Data cube and fx mask have same dims')
+            return True
+        else:
+            logger.error('Data cube and fx mask have different grids!')
+            sys.exit(1)
```

Comment:
That's the point, it should stop on error with traceback, but it just prints the error message and hangs. I think we should use `raise` instead of `sys.exit`.

Reply:
do you need this to be just a logger error (to stdout and screen) and the code should go on?

Reply:
It should stop.

Reply:
how is this not ```SystemExit: 1``` for you? ```sys.exit(1)``` stops all and exits the execution

Reply:
No idea, it just hangs without traceback.
@bouweandela ?

Reply:
what's the model that's hanging for? I tested with MPI-ESM-LR and MR and it works fine with the ```sftof``` files you gave me. ```sys.exit()``` propagates the failure and always gives you a stacktrace -- unless it's something not to like about your tcsh shell? what say you @bouweandela

Reply:
ow wait, you are running in multithread -- completely forgot about that, in this case ```os._exit()``` is needed pfff

Reply:
Just log the error and let the program crash, or raise a ValueError if you really want to abort execution yourself and things should work as expected.

Reply:
To explain a bit more: you should never call sys.exit() or similar from library code, but raise an exception instead, because the user of your code might want to handle that exception in their own way, maybe they do not want their program to exit.

Reply:
sounds good, I personally like exits (not brexits :grin:  ) but ok, the user should be the boss; sys.exit() doesn't play well with multithreading anyway, I'll just remove the ```sys.exit(1)``` calls and let the code die with a Traceback